code/globalincs/pool.h

```diff
@@ -0,0 +1,404 @@
+#pragma once
+
+#include "globalincs/pstypes.h"
+#include <tl/optional.hpp>
+#include <csignal>
+//#include <vector>
+
+using  tl::optional;
+struct pool_index{
```

Comment:
Can we call it SCP_pool_index, just to keep the naming consistent with the pool itself?

Reply:
for myself ergonomics overcomes consistency in this case. pool_index is already more than I'd like to be typing for this, honestly. Since it's going to be used all over the damn place in the end I'm more than willing to bend to consensus on this one, if one can be established.